.github/workflows/doc_builder.yml

```diff
@@ -0,0 +1,20 @@
+name: Publish docs via GitHub Pages
+on:
+  push:
+    branches:
+      - mainline
+    tags:        
+      - v*
```

Comment:
Is this an implicit AND, or will the workflow match on any push to mainline or any `v*` tag?

Reply:
good call. it is or logic so i get rid of the branch matching, since i guess we won't push tags to branches other than mainline

Reply:
refer to https://github.community/t/run-workflow-on-push-tag-on-specific-branch/17519